pkg/tfshim/sdk-v2/upgrade_state.go

```diff
@@ -103,8 +160,13 @@ func findID(v cty.Value) (string, bool) {
 	if !ok {
 		return "", false
 	}
-	if !id.Type().Equals(cty.String) {
+	if !id.Type().Equals(cty.String) || id.IsNull() {
```

Comment:
Interesting, I thought nulls in cty are typed, perhaps this accepts null that are not typed String.

Reply:
Null's in cty are typed, but `id.AsString()` where `id.IsNull()` panics. `findID` already returned `"", false` when it couldn't identify an accessible string ID. This just covers one more corner case.

Reply:
Ah that makes perfect sense. Same as the expanded form:


```
	if !id.Type().Equals(cty.String) {
		return "", false
	}
        if id.IsNull() {
  		return "", false
        }
	return id.AsString(), true
```